app/controllers/posts_controller.rb

```diff
@@ -33,6 +35,18 @@ def create
     end
   end
 
+  def destroy
+    @user = current_user
+    @post = @user.posts.find(params[:id])
+    authorize! :destroy, @post
```

Comment:
[OPTIONAL]
Hello @rubydevi, @yin-ka, @DianaBeki,

Great job on your `PostsController`! 🌟 Let's dive into the review:

### CanCanCan Authorization: 🔒
You've correctly included `load_and_authorize_resource` at the top of the `PostsController`, which ensures that CanCanCan handles the resource loading and authorization for each action. This is in line with the project requirements. 👍

However, there's a slight redundancy in your `destroy` action. You've manually called `authorize! :destroy, @post` even though `load_and_authorize_resource` already covers the authorization for this action. It's not necessarily harmful, but it's redundant. The power of `load_and_authorize_resource` is to avoid manual authorization checks like this in each action. 🔄

### Suggestions: 💡
1. **Remove Manual Authorization**: As mentioned, the manual call to `authorize!` is redundant due to `load_and_authorize_resource`. ❌

2. **Use of `before_action`**: To keep things DRY, we can use a `before_action` to set `@user` for relevant actions. 🧽

```ruby
class PostsController < ApplicationController
  load_and_authorize_resource
  before_action :set_current_user, only: [:new, :create, :destroy]

  ...

  def destroy
    @post = @user.posts.find(params[:id])

    if @post.destroy
      flash.now[:success] = "Your post titled: '#{@post.title}' was successfully deleted"
    else
      flash.now[:error] = 'Oops! Cannot delete your post.'
    end
    redirect_to user_posts_path(@user)
  end

  private

  def set_current_user
    @user = current_user
  end

  ...
end
```

By adding the `before_action`, you ensure that `@user` is set for the `new`, `create`, and `destroy` actions without repeating the same line of code. Additionally, by removing the redundant `authorize!` call in the `destroy` action, the code becomes cleaner and more readable. 📖

### Overall: 📊
Your implementation seems to be mostly in line with the project requirements. Address the mentioned suggestions to make your code cleaner and more efficient. 🛠️

Keep up the good work! 👏🚀

If you have any questions or need further clarification on any point, feel free to ask. 🙋 Remember to tag @topeogunleye so I'm notified promptly.

Best regards, 🤝
Temitope

Reply:
Thank you so much for your in detailed review. Your suggestions are a big help, and I appreciate it! @topeogunleye